windows.md

```diff
@@ -254,6 +254,7 @@ With those compatibility things out of the way, you're ready to start the system
     ```bash
     echo "export PATH=\$PATH:\"/c/Program Files/PostgreSQL/16/bin\"" >> "$USERPROFILE/.bash_profile"
     echo "export PGDATA=\"/c/Program Files/PostgreSQL/16/data\"" >> "$USERPROFILE/.bash_profile"
+    echo "export LC_ALL=en_US.UTF-8" >> "$USERPROFILE/.bash_profile"
```

Comment:
we still have a warning in your screenshot in the PR description:

```
WARNING: Console code page (850) differs from the Windows code page (1252). 8-bit characters might not work correctly. See psql reference page "Notes for Windows users" for details.
```

![Screenshot 2024-07-08 at 15 05 08](https://github.com/upleveled/system-setup/assets/1935696/4be33dd9-f516-476a-98de-20e815ec11be)

let's get rid of this warning while we're at it

the `psql reference page "Notes for Windows users"` in the error message may be referring to the page that @ProchaLu linked you:

- https://www.postgresql.org/docs/current/app-psql.html#:~:text=Notes%20for%20Windows%20Users

this page refers to [the `chcp` command](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/chcp), and recommends adding that to `/etc/profile` if on Cygwin (Cygwin is used internally in Git Bash)

we can however add it to `$USERPROFILE/.bash_profile` if we end up using `chcp` (make sure you check carefully if both `chcp` and `LC_ALL` are needed, for both of the issues):

- #73 
- #76

Reply:
So far, the only thing that has worked normally for this console warning is manually checking this box, as shown in the screenshot below. The `Beta: Use Unicode UTF-8 for worldwide language support` is not checked, and when you check it and restart it, the issue is gone.

![11111](https://github.com/upleveled/system-setup/assets/74430629/3cfe1b6c-6592-42f4-a5a6-0ab6a51bc6db)
![1111](https://github.com/upleveled/system-setup/assets/74430629/c73cb773-a055-43ef-953c-54ebc8bc479c)

Suggested solution in Docs

- Adding this `cmd.exe /c chcp 1252` to the `$USERPROFILE/.bash_profile`  : `echo "cmd.exe /c chcp 1252" >> "$USERPROFILE/.bash_profile"` => This did not work and also changed my command line (hyper) from this path style`Victor@Victor MINGW64~` to `C:\Users\Victor>`
- Adding `cmd.exe /c chcp 1252` to the `/etc/profile` file: This broke Postgres completely with this error message below
```bash
postgres weiß nicht, wo die Serverkonfigurationsdatei zu finden ist.
Sie müssen die Kommandozeilenoption --config-file oder -D angegeben oder
die Umgebungsvariable PGDATA setzen.
```
- It can be changed in PowerShell by running `chcp 1252` but this is only for the current session, when a new PowerShell is opened, it resets back to `850`, and running this `chcp 1252` in hyper throws `chcp command not found` error
- There are other suggested commands, but I wouldn't want to run all of them if I can't confirm exactly what it does
https://github.com/Disassembler0/Win10-Initial-Setup-Script/issues/252#:~:text=If%20you%27d%20really%20want%20to%20achieve%20the%20same%20as%20the%20checkbox%20does%2C%20you%20can%20create%20a%20custom%20tweak%20with%20following
https://stackoverflow.com/questions/56419639/what-does-beta-use-unicode-utf-8-for-worldwide-language-support-actually-do#:~:text=Windows%20Registry%20Editor%20Version
https://learn.microsoft.com/en-us/answers/questions/1041567/beta-use-unicode-utf-8-questions-windows-10?page=1


There are both official and none official suggestions that support the first approach
https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page#:~:text=nix%2Dbased%20platforms.-,Set%20a%20process%20code%20page%20to%20UTF%2D8,-As%20of%20Windows

https://stackoverflow.com/questions/57131654/using-utf-8-encoding-chcp-65001-in-command-prompt-windows-powershell-window

Reply:
if the warning leads to a rabbit hole, you can also move this out to another System Setup issue, for resolution in a different PR

Reply:
Moved this to a new issue
https://github.com/upleveled/system-setup/issues/83

Reply:
The `lc_messages` is responsible for displaying the messages in the `psql` console; when it is set to German `de_AT.UTF-8`,  then the messages like error/success output will be in German. E.g `FEHLER:  Relation »nothing« existiert nicht`.

Even though the `LC_MESSAGES` is set to English in this output below, it doesn't affect `lc_messages` value because `LC_MESSAGES` is `Environment Variable`  while `lc_messages` is `PostgreSQL Configuration Parameter` that lives inside the `postgresql.conf` file.

```bash
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=en_US.UTF-8
```

Testing in Linux

`SHOW lc_messages`

Displays output in German
```bash
 lc_messages 
-------------
 de_AT.UTF-8
```

Displays output in English
```bash
lc_messages 
-------------
 en_US.UTF-8
```

Testing on Windows

`SHOW lc_messages`

Two windows tested display different outputs but indicate all is set to the English language, as shown below

```bash
----------------------
 English_Austria.1252
(1 row)
```
```bash
lc_messages
-----------------------------
 English_United Kingdom.1252
```
When `lc_messages` was set back to `de_AT.UTF-8`, it resulted back to German display outputs

This `lc_messages` needs to be set to `en_US.UTF-8` inside the  `postgresql.conf` file in order to make the setting persist across postgres sessions